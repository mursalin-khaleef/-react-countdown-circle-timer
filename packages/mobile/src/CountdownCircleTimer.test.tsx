import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { Text } from 'react-native'
import { render } from '@testing-library/react-native'
import { renderHook } from '@testing-library/react-hooks'
import renderer from 'react-test-renderer'

import { CountdownCircleTimer, useCountdown } from '.'
import type { Props } from '.'

const fixture: Props = {
  duration: 0.2,
  colors: '#abc',
}

describe('CountdownCircleTimer', () => {
  it('renders with single color', () => {
    const tree = renderer
      .create(
        <CountdownCircleTimer {...fixture}>
          {({ remainingTime }) => <Text>{remainingTime}</Text>}
        </CountdownCircleTimer>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders with different trail stroke width', () => {
    const tree = renderer
      .create(
        <CountdownCircleTimer
          {...fixture}
          trailStrokeWidth={16}
          strokeWidth={14}
        >
          {({ remainingTime }) => <Text>{remainingTime}</Text>}
        </CountdownCircleTimer>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('starts the timer from value provided in initialRemainingTime', () => {
    const { getByText } = render(
      <CountdownCircleTimer
        duration={10}
        colors="#abc"
        initialRemainingTime={3.7}
      >
        {({ remainingTime }) => <Text>{remainingTime}</Text>}
      </CountdownCircleTimer>
    )

    expect(getByText('4')).toBeTruthy()
  })

  it('does not render the animating path when the elapsed time is equal the duration', () => {
    const tree = renderer.create(
      <CountdownCircleTimer
        duration={0}
        colors="#abc"
        initialRemainingTime={3.7}
      >
        {({ remainingTime }) => <Text>{remainingTime}</Text>}
      </CountdownCircleTimer>
    )

    expect(tree).toMatchSnapshot()
  })
})

describe('useCountdown', () => {
  it('returns default values form useCountdown', () => {
    const { result } = renderHook(() =>
      useCountdown({ duration: 12, colors: '#f1f2f3' })
    )

    expect(result.current).toEqual({
      elapsedTime: 0,
      path: 'm 90,6 a 84,84 0 1,0 0,168 a 84,84 0 1,0 0,-168',
      pathLength: 527.7875658030853,
      remainingTime: 12,
      rotation: 'clockwise',
      size: 180,
      stroke: '#f1f2f3',
      strokeDashoffset: 0,
      strokeWidth: 12,
    })
  })
})
