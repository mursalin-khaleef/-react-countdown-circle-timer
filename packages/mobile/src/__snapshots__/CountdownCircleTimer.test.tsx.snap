// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountdownCircleTimer does not render the animating path when the elapsed time is equal the duration 1`] = `
<View
  style={
    Object {
      "height": 180,
      "position": "relative",
      "width": 180,
    }
  }
>
  <RNSVGSvgView
    bbHeight={180}
    bbWidth={180}
    focusable={false}
    height={180}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 180,
          "width": 180,
        },
      ]
    }
    width={180}
  >
    <RNSVGGroup>
      <RNSVGPath
        d="m 90,6 a 84,84 0 1,0 0,168 a 84,84 0 1,0 0,-168"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={4292467161}
        strokeWidth={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text>
      0
    </Text>
  </View>
</View>
`;

exports[`CountdownCircleTimer renders with different trail stroke width 1`] = `
<View
  style={
    Object {
      "height": 180,
      "position": "relative",
      "width": 180,
    }
  }
>
  <RNSVGSvgView
    bbHeight={180}
    bbWidth={180}
    focusable={false}
    height={180}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 180,
          "width": 180,
        },
      ]
    }
    width={180}
  >
    <RNSVGGroup>
      <RNSVGPath
        d="m 90,8 a 82,82 0 1,0 0,164 a 82,82 0 1,0 0,-164"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={4292467161}
        strokeWidth={16}
      />
      <RNSVGPath
        d="m 90,8 a 82,82 0 1,0 0,164 a 82,82 0 1,0 0,-164"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
          ]
        }
        stroke={4289379276}
        strokeDasharray={
          Array [
            515.221195188726,
            515.221195188726,
          ]
        }
        strokeDashoffset={null}
        strokeLinecap={1}
        strokeWidth={14}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text>
      1
    </Text>
  </View>
</View>
`;

exports[`CountdownCircleTimer renders with single color 1`] = `
<View
  style={
    Object {
      "height": 180,
      "position": "relative",
      "width": 180,
    }
  }
>
  <RNSVGSvgView
    bbHeight={180}
    bbWidth={180}
    focusable={false}
    height={180}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 180,
          "width": 180,
        },
      ]
    }
    width={180}
  >
    <RNSVGGroup>
      <RNSVGPath
        d="m 90,6 a 84,84 0 1,0 0,168 a 84,84 0 1,0 0,-168"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
          ]
        }
        stroke={4292467161}
        strokeWidth={12}
      />
      <RNSVGPath
        d="m 90,6 a 84,84 0 1,0 0,168 a 84,84 0 1,0 0,-168"
        fill={null}
        propList={
          Array [
            "fill",
            "stroke",
            "strokeWidth",
            "strokeDasharray",
            "strokeDashoffset",
            "strokeLinecap",
          ]
        }
        stroke={4289379276}
        strokeDasharray={
          Array [
            527.7875658030853,
            527.7875658030853,
          ]
        }
        strokeDashoffset={null}
        strokeLinecap={1}
        strokeWidth={12}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Text>
      1
    </Text>
  </View>
</View>
`;
